CHERI-RISC-V CLoadTags support

nwf opened the pull request.

nwf-msr committed:
CHERI-RISC-V CLoadTags support

arichardson commented on target/riscv/insn_trans/trans_cheri.c.inc:
```diff
@@ -164,6 +164,8 @@ TRANSLATE_CGET(cgettag)
 TRANSLATE_CGET(cgettype)
 TRANSLATE_CGET(cgetsealed)
 
+TRANSLATE_CGET(cloadtags)
```
We might want to consider renaming `TRANSLATE_CGET` -> `TRANSLATE_INT_CAP` now that it is used by non-`cget*`-instructions. And the same for `gen_cheri_get`.

nwf replied:
Done.

arichardson commented on target/riscv/op_helper_cheri.c:
```diff
@@ -443,3 +444,27 @@ target_ulong HELPER(sc_c_cap)(CPUArchState *env, uint32_t addr_reg, uint32_t val
+    } else if (!(cbp->cr_perms & CAP_PERM_LOAD)) {
+        raise_cheri_exception(env, CapEx_PermitLoadViolation, cb);
+    } else if (!(cbp->cr_perms & CAP_PERM_LOAD_CAP)) {
+        raise_cheri_exception(env, CapEx_PermitLoadCapViolation, cb);
```
Other loads are tag-stripping in this case. I know it's not useful for cloadtags, but it would match Morello.
Maybe worth adding a comment such as "CLoadTags raises an exception instead of stripping tags"?

arichardson reviewed (approved):
Haven't check that the encoding matches, but QEMU code looks good to me.

nwf-msr committed:
CHERI-RISCV: TRANSLATE_CGET -> TRANSLATE_INT_CAP

We're about to use this for CLoadTags, too, so it's not just for CGet* any more.
